### PR TITLE
[INT-201] #CheckWithTech enhancements

### PR DESCRIPTION
**Pull request by markspolakovs**
- Stores the state of requests in the database, and displays on events to techies
- Full Slack integration
  - Techies can action requests from within Slack
  - Requestors are notified of the status

TODO:

- [ ] We should probably warn if the requester isn't signed in to Slack so the approver lets them know some other way
- [ ] An admin UI to list pending CWTs

### CODE REVIEW

File: features/calendar/events.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import "server-only";
+// import "server-only";
```
Comment (markspolakovs) [inline features/calendar/events.ts:1]:
```suggestion
import "server-only";
```

File: lib/env.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ const envSchema = z.object({
   SLACK_CLIENT_ID: slackEnvType,
   SLACK_CLIENT_SECRET: slackEnvType,
   SLACK_TEAM_ID: z.string().optional(),
-  SLACK_CHECK_WITH_TECH_CHANNEL: slackEnvType.default("#check-with-tech"),
+  SLACK_CHECK_WITH_TECH_CHANNEL: slackEnvType,
```
Comment (markspolakovs) [inline lib/env.ts:39]: ACHTUNG: there's some places that the slack API requires a channel ID, so I remove the default here. We need to make sure the ID is set in ystv/nomad prior to merging.

Comment (markspolakovs) [reply lib/env.ts:39]: Done.

File: features/calendar/check_with_tech_actions.ts [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+      identities: {
+        some: {
+          provider: "slack",
+          provider_key: actorId,
+        },
+      },
+    },
+  });
+  if (!actor) {
+    await respond({
```
Comment (archessmn) [inline features/calendar/check_with_tech_actions.ts:57]: Can we have this message only show for the user that clicked it?

File: features/calendar/check_with_tech_actions.ts [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+      text: `Please <${env.PUBLIC_URL}/user/me|link your Internal Site profile to Slack> to use this feature.`,
+    });
+    return;
+  }
+  if (!userHasPermission(actor.user_id, "CheckWithTech.Admin")) {
+    await respond({
```
Comment (archessmn) [inline features/calendar/check_with_tech_actions.ts:63]: Same with only showing to the single user

**Review by archessmn — CHANGES_REQUESTED** [review]
Would be good to have these bits changed as well as having the "need help" flow use it as well.

I've made a few UI changes but otherwise looks good.

File: features/calendar/check_with_tech_actions.ts (hunk 3 above)
Comment (markspolakovs) [reply features/calendar/check_with_tech_actions.ts:57]: Fairly sure that this should be an ephemeral message, per https://api.slack.com/interactivity/handling#publishing_ephemeral_response

Comment (archessmn) [reply features/calendar/check_with_tech_actions.ts:57]: I thought that's what it was but couldn't remember if it was something different for slack

File: features/calendar/check_with_tech_actions.ts (hunk 4 above)
Comment (markspolakovs) [reply features/calendar/check_with_tech_actions.ts:63]: Same response

**Comment by markspolakovs** [comment]
image

File: docs/setup_slack.md [hunk 5]
```diff
@@ -51,6 +51,16 @@ Sroll down slightly to `User Token Scopes` and add `team.read`.
 
 You can now scroll up to the top of the page and click `Install to Workspace`. Click `Allow` when prompted. This should take you back to the **OAuth & Permissions** page. Towards the top of the page you should see `Bot User OAuth Token`. Copy the value of this over to `SLACK_BOT_TOKEN`.
 
+## Setting up channels
+
+This app uses a number of slack channels for integration with check-with-tech responses, feedback, and others. These three channels are:
+
+- `SLACK_CHECK_WITH_TECH_CHANNEL` - Used for check-with-tech requests
+- `SLACK_TECH_HELP_CHANNEL` - Used for help with tech requests
```
Comment (markspolakovs) [inline docs/setup_slack.md:59]:
```suggestion
```
rip

File: features/calendar/check_with_tech_actions.ts (hunk 3 above)
Comment (archessmn) [reply features/calendar/check_with_tech_actions.ts:58]: I've got two accounts on my dev instance and can see the messages on both

**Comment by archessmn** [comment]
image
Needs spacing fixing

**Review by archessmn — APPROVED** [review]

File: lib/auth/permissions.ts [hunk 6]
```diff
@@ -22,6 +22,8 @@ export const PermissionEnum = z.enum([
   "Calendar.Public.Admin",
   "Calendar.Public.Creator",
   "CalendarIntegration.Admin",
+  "CheckWithTech.Submit",
```
[the remark this answers is not in the document]
Comment (markspolakovs) [reply lib/auth/permissions.ts:25]: And doesn't show the prompt at all - essentially intended as a feature flag.

**Pull request merged**